source-zendesk-support-native/source_zendesk_support_native/api.py

```diff
@@ -313,23 +313,6 @@ async def backfill_satisfaction_ratings(
     yield end
 
 
-async def _fetch_top_level_fields(
-    http: HTTPSession,
-    url: str,
-    params: dict[str, str | int],
-    response_model: type[IncrementalCursorExportResponse],
-    log: Logger,
-) -> tuple[str | None, bool]:
-    # Instead of using Pydantic's model_validate_json that uses json.loads internally,
-    # use json.JSONDecoder().raw_decode to reduce memory overhead when processing the response.
-    raw_response_bytes = await http.request(log, url, params=params)
```

Comment:
This would previously require buffering the entire response's bytes and plucking out a couple values from it, then making the same request again to incrementally process. Now we will only need to make the request a single time while still being able to incrementally process it.